### PR TITLE
HLTTauDQM: ignoring modules starting with '-' (modules inside cms.ign…

### PR DESCRIPTION
**Pull request by slehti**
New DeepTau paths contain modules inside cms.ignore(). This caused problems with the offline object matching, HLTTauDQM stopped when not finding offline muons for di-tau path (inside ignore there are all L1 seeds including mu+tau seed, therefore the code is looking for muons, too).
Fix: ignore all modules inside cms.ignore(): i.e. modules starting with a '-'. 

Files changed: DQMOffline/Trigger/src/HLTTauDQMPath.cc

### CODE REVIEW

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
+1

File: DQMOffline/Trigger/src/HLTTauDQMPath.cc [hunk 1]
```diff
@@ -54,6 +54,8 @@ namespace {
           continue;
         if (type == "HLTTriggerTypeFilter" || type == "HLTBool")
           continue;
+        if (iLabel->find('-') == 0)  // ignore
```
Comment (perrotta) [inline DQMOffline/Trigger/src/HLTTauDQMPath.cc:57]: Aren't there modules with a "-" somewhere in the name? Wouldn't it be safer to check that they just don't start with a "-"? 
(perhaps @cms-sw/hlt-l2 could remind me if a "-" in the name is just forbidden for hlt modules, in which case the proposed syntax is ok)

Comment (missirol) [reply DQMOffline/Trigger/src/HLTTauDQMPath.cc:57]: Hi @perrotta 

> Wouldn't it be safer to check that they just don't start with a "-"?

I think that's what this check is doing: it checks if character `-` is at position `0` using `find`, and if so it skips the word.

> Aren't there modules with a "-" somewhere in the name?

If you mean a `-` somewhere in the middle of the module name/label, as far as I know that's not possible for any module, not just HLT (that works for `_`, of course, but not for other special characters like `-`).

**Comment by perrotta** [comment]
+1

**Pull request merged**